langlib/lang.function/src/main/java/org/ballerinalang/langlib/function/Call.java

```diff
@@ -0,0 +1,134 @@
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.langlib.function;
+
+import io.ballerina.runtime.api.async.StrandMetadata;
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.types.Parameter;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.values.BFunctionPointer;
+import io.ballerina.runtime.internal.TypeChecker;
+import io.ballerina.runtime.internal.scheduling.Scheduler;
+import io.ballerina.runtime.internal.types.BArrayType;
+import io.ballerina.runtime.internal.types.BFunctionType;
+import io.ballerina.runtime.internal.types.BTupleType;
+import io.ballerina.runtime.internal.util.exceptions.BLangExceptionHelper;
+import io.ballerina.runtime.internal.util.exceptions.RuntimeErrors;
+import io.ballerina.runtime.internal.values.ArrayValueImpl;
+import io.ballerina.runtime.internal.values.ListInitialValueEntry;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import static io.ballerina.runtime.api.constants.RuntimeConstants.BALLERINA_BUILTIN_PKG_PREFIX;
+import static io.ballerina.runtime.api.constants.RuntimeConstants.FUNCTION_LANG_LIB;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.INCOMPATIBLE_TYPES;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.getModulePrefixedReason;
+
+/**
+ * Native implementation of lang.function:call(function func, any|error... args).
+ *
+ * @since 2201.2.0
+ */
+public class Call {
+
+    private static final StrandMetadata METADATA = new StrandMetadata(BALLERINA_BUILTIN_PKG_PREFIX, FUNCTION_LANG_LIB,
+                                                                      "1.0.0", "call");
+
+    public static Object call(BFunctionPointer<Object, Object> func, Object... args) {
+        BFunctionType functionType = (BFunctionType) func.getType();
+        List<Type> paramTypes = new LinkedList<>();
+        List<Type> argTypes = new LinkedList<>();
+        List<Object> argsList = new java.util.ArrayList<>(List.of(Scheduler.getStrand()));
+
+        if (validatePositionalArgs(args, argsList, functionType, paramTypes, argTypes) ||
+                validateRestArgs(args, argsList, functionType, paramTypes, argTypes)) {
+                throw ErrorCreator.createError(
+                        getModulePrefixedReason(FUNCTION_LANG_LIB, INCOMPATIBLE_TYPES),
+                        BLangExceptionHelper.getErrorDetails(RuntimeErrors.INCOMPATIBLE_TYPE,
+                                                             new BTupleType(paramTypes), new BTupleType(argTypes)));
+        }
+
+        return func.asyncCall(argsList.toArray(), METADATA);
+    }
+
+    private static boolean validatePositionalArgs(Object[] args, List<Object> argsList, BFunctionType functionType,
```

Comment:
IMO the natural thing for a function that starts with validate would be to return true if valid. Shall we change the method name or flip the check?

Reply:
Change the method name. please check.